joystick.c

```diff
@@ -0,0 +1,178 @@
+/**********************************************/
+// File     :     joystick.c
+// Author   :     John Bliss
+// Date     :     September 27th 2019
+/**********************************************/
+
+#include "joystick.h"
+
+
+enum joy_status joy1_mode = NONE;
+enum joy_status joy2_mode = NONE;
+
+
+static SDL_GameController *joystick1 = NULL;
+static SDL_GameController *joystick2 = NULL;
+static bool old_clock = false;
+static bool writing = false;
+static uint16_t joystick1_state = 0;
+static uint16_t joystick2_state = 0;
+static uint8_t clock_count = 0;
+
+
+bool joystick_latch, joystick_clock;
+bool joystick1_data, joystick2_data;
+
+bool joystick_init()
+{
+	int joystick1_number = 0;
+	//Try to get first controller, if it is not set to 1
+	if (joy1_mode != NONE) {
+		for (int i = 0; i < SDL_NumJoysticks(); i++) {
+			if (SDL_IsGameController(i)) {
+				joystick1 = SDL_GameControllerOpen(i);
+				if (joystick1) {
+					joystick1_number = i;
+					break;
+				} else {
+					fprintf(stderr, "Could not open gamecontroller %i: %s\n", i, SDL_GetError());
+				}
+			}
+		}
+	}
+	if (joy2_mode != NONE) {
+		for (int i=0; i < SDL_NumJoysticks(); i++) {
+			if (SDL_IsGameController(i) && joystick1_number != i) {
+				joystick2 = SDL_GameControllerOpen(i);
+				if (joystick2) {
+					break;
+				} else {
+					fprintf(stderr, "Could not open gamecontroller %i: %s\n", i, SDL_GetError());
+				}
+			}
+		}
+	}
+	writing = false;
+	return true;
+}
+
+void joystick_step()
+{
+	if (joy1_mode == NONE) {
+		return;
```

Comment:
That code doesn't let us use _only_ `-joy2`!  (We might want the keyboard to be JOY1.)

Reply:
I've done some updates to make sure that joy2 can work at the same time as the keyboard now.  It turns out I wasn't even properly checking joy2_mode.